aiohttp/client_ws.py

```diff
@@ -377,12 +377,20 @@ async def receive(self, timeout: Optional[float] = None) -> WSMessage:
 
     async def receive_str(self, *, timeout: Optional[float] = None) -> str:
         msg = await self.receive(timeout)
+        if msg.type not in MESSAGE_TYPES_WITH_CONTENT:
```

Comment:
I don't think the issue is related to content, only a TEXT message will work here. The issue was just asking for the TypeError to be updated in the below check.

Reply:
in the original issue, according to my understanding, it was observed that these `receive_*` methods were raising `TypeError` when `EofStream` exception was caught in the `receive` method:
https://github.com/aio-libs/aiohttp/blob/d74cb5cd1221759c80cb84141ac042b7eb765a7f/aiohttp/client_ws.py#L342-L346

I understood that in the issue, it was proposed that some exception other than `TypeError` should be raised for these cases, as these seem to be different from type conversions. 
Please correct me if you find anything wrong with my understanding.

Reply:
Consequentially, I think that checking for `MESSAGE_TYPES_WITH_CONTENT` provides a more generic implementation for catching such cases.

Reply:
Hmm, I'm not sure anymore. I was focusing more on:

> I would raise RuntimeError or so if corresponding message "type" (like bytes, str, or json) can not be received.

Which suggests changing the TypeError being thrown below.

And:

> The webscocket.receeive_* functions should throw ConnectionResetError when there are connection related errors instead of TypeError.

Which does make more sense to me than a RuntimeError, as people are likely already handling connection errors.

bdraco is more familiar with websockets, so I'll leave with him to decide what should be done.